sknn/backend/lasagne/mlp.py

```diff
@@ -276,6 +276,10 @@ def _array_to_mlp(self, array, nn):
         for layer, data in zip(nn, array):
             if data is None: continue
             weights, biases = data
+            if weights.dtype == numpy.float64:
+                weights = weights.astype(numpy.float32)
```

Comment:
Can you get the correct dtype from `layer.W` or use `floatX()` from Theano instead? This would avoid using the conditional, you can just always cast each time.